Update Squirrel & NuGet

fcastilloec opened the pull request:
This PR updates both Squirrel and NuGet to their latest versions, which has some bugs fixed. Both are only minor updates, so there aren't any breaking changes to our code.

This week I'll also try to update all the dependencies to their latest version in a different PR, so if we release a new version of this module, it should be after that second PR.
I'll also try fixing the links so they point to `electron-userland`.

The latest version of Squirrel was downloaded from https://www.nuget.org/packages/squirrel.windows
The latest version of Nuget was downloaded from https://www.nuget.org/downloads

malept commented on .travis.yml:
```diff
@@ -14,6 +14,7 @@ deploy:
     tags: true
 
 env:
+  - DEBUG="electron-installer-windows" NODE_VERSION="10"
```
Also for `appveyor.yml`?

malept replied:
Or perhaps we should try using Travis Windows support.

malept replied:
> I'm just going to add support to AppVeyor

I added it: https://ci.appveyor.com/project/electron-userland/electron-installer-windows

I can add you to the AppVeyor org, I just need to figure out how to send an invitation.

> and in the future, I'll investigate Travis Windows and see if it's better than using AppVeyor.

I tried it with `electron-installer-zip` the other week (also in electron-userland) and it was hanging for some reason. The forums said it was because there were processes still running, so your mileage may vary.

fcastilloec replied:
To send me an invite, go to `Settings` → `Team`

fcastilloec commented:
It's expected that the executables change when a new Squirrel version is released. I'm not 100% sure how they all interact with each other, but each new release comes with all those tools.
The Github release page used to contain a `.zip` which everything necessary to use Squirrel, which is exactly what the `vendor/squirrel` directory contains. Nowadays, we need to extract the contents from the raw `.nupkg` file found in the NuGet Gallery's link in the description.

Regarding the `.exe` files, I know that we use `Squirrel.exe` or `Squirrel-Mono.exe` depending on the OS we're using (Windows vs. Linux/macOS). `Setup.exe` is one of the executables added to the release that deals with installation and updating of the app. `SyncReleases.exe` deals with checking for new releases and it's also added to the final `.exe`. I'm not very sure what `WriteZipToSetup.exe` does, but I know that Squirrel requires some type of compression and it's usually `7z` which is also included in every release but it doesn't get updated very often, I'm guessing `WriteZipToSetup.exe` might have a similar.

@malept I also noticed that I forgot about AppVeyor orginially because it seems that it's not configured in this repo anymore since we moved to `electron-userland`. I used to have access to the repo in Appveyor, mainly to restart failed builds but because of the move, that's not longer working.
If you have the admin rights, please add the AppVeyor app to this repo and, if possible, add me as a [Collaborator](https://www.appveyor.com/docs/team-setup/#collaborator) (my email: fcastillo.ec(at)gmail.com)

malept commented:
@fcastilloec I've sent you an invitation to join the `electron-userland` AppVeyor org. Perhaps the next steps are:

1. Make sure you have access to AppVeyor
2. Separate PR to update the badge and get it merged
3. Rebase this PR on master so AppVeyor gets triggered

fcastilloec commented:
@malept I have access to AppVeyor now, thanks! I can submit a new PR updating the badge to test that AppVeyor is being triggered and then rebase this PR after the badge-PR gets merged.

fcastilloec commented:
@malept It seems AppVeyor and GitHub don't want to play nice with each other right now. I've noticed that if a build fails, and I restart it, then the status gets stuck at "Pending...", even after restarted it passes.
I've reached to AppVeyor on Twitter to see if they have something to say about this.

fcastilloec commented:
@malept AppVeyor replied to the issue. Because I'm not an administrator on AppVeyor or GitHub, I can't do what they're asking me to do. Here's the [tweet](https://twitter.com/appveyor/status/1067640870394986496). I think it has to do with migrating to `electron-userland`, so deleting all authentications and enabling the GitHub app might do the trick.
